internal/go_repository_config.bzl

```diff
@@ -0,0 +1,102 @@
+# Copyright 2014 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_go//go/private:common.bzl", "env_execute", "executable_extension")
+load("@bazel_gazelle//internal:go_repository_cache.bzl", "read_cache_env")
+
+def _go_repository_config_(ctx):
+    # Locate and resolve configuration files. Gazelle reads directives and
+    # known repositories from these files. Resolving them here forces
+    # go_repository rules to be invalidated when they change. Gazelle's cache
+    # should NOT be invalidated, so we shouldn't need to download these again.
+    # TODO(#549): vcs repositories are not cached and still need to be fetched.
```

Comment:
Let's keep this TODO in `_go_repository_impl`. The problem is that we're not caching vcs repositories. We wouldn't implement that cache here.